From: Prarit Bhargava <prarit@redhat.com>
Date: Tue, 9 Dec 2008 10:09:53 -0500
Subject: [acpi] add xw8600 and xw6600 to GPE0 block blacklist
Message-id: 20081209151004.26311.44743.sendpatchset@prarit.bos.redhat.com
O-Subject: [RHEL 5.4 PATCH]: Add xw8600 and xw6600 to GPE0 block blacklist
Bugzilla: 475418
RH-Acked-by: Neil Horman <nhorman@redhat.com>
RH-Acked-by: Brian Maly <bmaly@redhat.com>

HP xw6800 and xw8800 systems have strange Product Names in their DMI tables.

Add them to the GPE0 address register blacklist.

Tested on HP xw6800 (during testing I hit another unrelated issue) -- but
the patch does resolve the GPE0 register errors.

Resolves BZ 475418.

diff --git a/drivers/acpi/events/evgpeblk.c b/drivers/acpi/events/evgpeblk.c
index 1b9852f..8ba2128 100644
--- a/drivers/acpi/events/evgpeblk.c
+++ b/drivers/acpi/events/evgpeblk.c
@@ -1082,6 +1082,22 @@ static struct dmi_system_id acpi_hp_gpe_table[] = {
 		     DMI_MATCH(DMI_PRODUCT_NAME, "HP xw"),
 	 	    },
 	},
+	{
+	 .callback = hp_gpe_use_32_bit,
+	 .ident = "HP me09 Workstation",
+	 .matches = {
+		     DMI_MATCH(DMI_BOARD_VENDOR, "Hewlett-Packard"),
+		     DMI_MATCH(DMI_PRODUCT_NAME, "HP me09 Workstation"),
+	 	    },
+	},
+	{
+	 .callback = hp_gpe_use_32_bit,
+	 .ident = "HP he09 Workstation",
+	 .matches = {
+		     DMI_MATCH(DMI_BOARD_VENDOR, "Hewlett-Packard"),
+		     DMI_MATCH(DMI_PRODUCT_NAME, "HP he09 Workstation"),
+	 	    },
+	},
 	 {}
 };
 
